app/js/lib/foxbox.js

```diff
@@ -40,54 +40,82 @@ let themes = [
 
 const _private = {
   isPollingEnabled: Symbol('isPollingEnabled'),
-  nextPollTimeout: Symbol('nextPollTimeout')
+  nextPollTimeout: Symbol('nextPollTimeout'),
+
+  fetchServices: Symbol('fetchServices')
 };
 
 /**
- * Request a JSON from a specified URL.
+ * Request a content of the specified type from a specified URL.
  *
- * @param {string} url The URL to send the request to.
- * @param {string} method The HTTP method (defaults to "GET").
- * @param {!Object} body An object of key/value.
+ * @param {String} url The URL to send the request to.
+ * @param {String} contentType The content mime type (eg. image/jpeg).
+ * @param {String} [method=GET] The HTTP method (defaults to "GET").
+ * @param {Object} [body] An object of key/value.
  * @return {Promise}
  */
-const fetchJSON = function(url, method = 'GET', body = undefined) {
+const fetchContent = function(url, contentType, method = 'GET', body = null) {
   method = method.toUpperCase();
 
   const req = {
     method,
-    headers: {
-      'Accept': 'application/json'
-    },
-    cache: 'no-store'
+    headers: { Accept: contentType },
+    cache: 'no-store',
+    body: body && JSON.stringify(body)
   };
 
   if (method === 'POST' || method === 'PUT') {
     req.headers['Content-Type'] = 'application/json;charset=UTF-8';
   }
+
   if (settings.session) {
     // The user is logged in, we authenticate the request.
     req.headers.Authorization = `Bearer ${settings.session}`;
   }
 
-  if (method === 'GET' || method === 'HEAD') {
-    body = undefined;
-  }
-  if (body !== undefined) {
-    req.body = JSON.stringify(body);
-  }
-
   return fetch(url, req)
     .then(res => {
-      if (res.ok) {
-        return res.json();
+      if (!res.ok) {
+        throw new TypeError(
+          `The response returned a ${res.status} HTTP status code.`
+        );
       }
 
-      throw new TypeError(`The response returned a ${res.status} HTTP status code.`);
+      return res;
+    }).catch((e) => {
+      console.error('Error occurred while fetching content: ', e);
+      throw e;
```

Comment:
I don't think we should catch here, otherwise the application won't have any ways to know whether the request was successful or not.
It's up to the app to append `catch()` to the promise returned by the request.